Text Generation WebUI/scripts/generate-config-schema-from-csv.cjs

```diff
@@ -0,0 +1,298 @@
+'use strict';
+const assert = require('assert/strict');
+const fs = require('fs');
+const { type } = require('os');
+
+const csvFilePath = process.argv[2];
+if (!csvFilePath) {
+  throw 'Specify csv file path as parameter';
+}
+
+const headersObj = {
+  category: 'Category',
+  group: 'Group',
+  subgroup: 'Subgroup',
+  name: 'Name',
+  variable: 'Variable',
+  cmdParameter: 'cmd parameter',
+  type: 'Type',
+  valueType: 'ValueType',
+  options: 'Options',
+  useConditions: 'Use Condition',
+  default: 'Default Value',
+  min: 'Min Value',
+  max: 'Max Value',
+  description: 'Description',
+};
+
+const types = {
+  category: 0,
+  group: 1,
+  subgroup: 2,
+  number: 3,
+  text: 4,
+  textarea: 5,
+  checkbox: 6,
+  radio: 7,
+  select: 8,
+  multiselect: 9,
+  slider: 10,
+};
+
+const valueType = {
+  default_type: 0,
+  double: 1,
+  float: 2,
+  int32: 3,
+  int64: 4,
+  uint32: 5,
+  uint64: 6,
+  sint32: 7,
+  sint64: 8,
+  fixed32: 9,
+  fixed64: 10,
+  sfixed32: 11,
+  sfixed64: 12,
+  bool: 13,
+  string: 14,
+  bytes: 15,
+};
+
+const splitByComma = (str) => {
+  const result = [];
+  let current = '';
+  let insideQuotes = false;
+
+  for (let i = 0; i < str.length; i++) {
+    const char = str[i];
+
+    if (char === '"') {
+      insideQuotes = !insideQuotes; // Toggle the insideQuotes flag
+    }
+
+    if (char === ',' && !insideQuotes) {
+      result.push(current.trim());
+      current = '';
+    } else if (char !== '"') {
+      current += char;
+    }
+  }
+
+  // Add the last part
+  result.push(current.trim());
+
+  return result;
+};
+
+const parseCsv = (data) => {
```

Comment:
Why don't you use parser libraries?

Reply:
This is a self-sufficient script. It will be useful not only for the Text Generation WebUI but also for anything that requires creating a configuration schema. We will be porting it to other engines, which is why I didn’t want to tie it to the package.json.